bin/main

```diff
@@ -1,2 +1,55 @@
 #!/usr/bin/env ruby
-puts 'Hello, world! '
+
+# get the player1 name
+puts 'Hey player1, whats your name?'
+player1 = gets.chomp
+puts "hello #{player1}, Please select a token (x / o)"
+# get the player1 token (x / o)
+player1_token = gets.chomp
+puts "#{player1_token}, Nice choice!"
+
+# get the player2 name
+puts 'Hey player2, whats your name?'
+player2 = gets.chomp
+# get the player2 token (x / o)
+player2_token = player1_token == 'x' ? 'o' : 'x'
+puts "hello #{player2}, your token is #{player2_token}"
+# display the board
+def display_board
+  board = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+  p board[0]
+  p board[1]
+  p board[2]
+end
+# create a loop to take each player move
+puts 'select a numberd position from the board'
+display_board
+game_over = false
+until game_over
+  # get the player1 move
+  puts "#{player1}, choose a position"
+  player1_position = gets.chomp
+  # check if the position is valid
+  # confirm selected move
+  puts "You selected position #{player1_position}."
+  # display the board
+  display_board
+  # check if the game is complete
+  # return a winner
+  # else
+  # next
+  # get the player2 move
+  puts "#{player2}, choose a position"
+  player2_position = gets.chomp
+  # check if the position is valid
+  # confirm selected move
+  puts "You selected position #{player2_position}."
+  # display the board
+  display_board
+  # check if the game is complete
+  # return a winner
+  puts "Hurray, #{player2} you win"
```

Comment:
- [ ] Notice that you are repeating the same process for each player's turn. Consider extracting this as a method you would call in the loop for each player turn.

Reply:
> * [ ]  Notice that you are repeating the same process for each player's turn. Consider extracting this as a method you would call in the loop for each player turn.

Please consider extracting the repeated code in a method like this and calling it in each player's turn within the loop:
```
def each_turn(player)
  puts "#{player}, choose a position"
  player_position = gets.chomp
  # check if the position is valid
  # confirm selected move
  puts "You selected position #{player_position}."
  # display the board
  display_board
end
```